### DISCUSSION TITLE
Disable implicit behaviour for query parameter format

### DISCUSSION THREAD

**Discussion by tu-maurice** [mojolicious/mojo#1760] [Q&A] [answered]
Is there a way to disable implicit behaviour in Mojo involving the query/get parameter `format`? My use case involves supplying every single get parameter to a query in a backend. That query does involve a field called format that would be supplied to my API as a get parameter. Now when specifying the format parameter Mojo tries to derive something from that although I just want my application to forward that key value pair to the backend.

I've tried to explicitly set the response format as follows, but a request to `/search.json?format=html` still ends up with a html response instead of a json respone.

```
  $r->get('/search.json')->to(
    controller => "search",
    action => "search",
    format => "json",
  )->name('search');

  $r->get('/search.html')->to(
    controller => "search",
    action => "search",
    format => "html",
  )->name('search');
```

**Comment by kraih** [comment, marked as answer]
I'm afraid not. In retrospect, we probably should have used `_format`, similar to how we use `_method`.

**Reply by tu-maurice** [reply]
Too bad.

**Reply by kraih** [reply]
We've deprecated `?format=*`. https://github.com/mojolicious/mojo/compare/eeef1e2a52fd...c9588f33fbc2
